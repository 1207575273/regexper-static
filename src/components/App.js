import React from 'react';

import Message from 'components/Message';

const App = () => (
  <Message heading="React App">
    <p>Placeholder app content</p>
  </Message>
);

export default App;
