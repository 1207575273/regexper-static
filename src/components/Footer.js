import React from 'react';

const Footer = () => (
  <footer>
    <ul className="inline with-separator">
      <li>
        Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a>
      </li>
      <li>
        Generated images licensed: <a rel="license" href="http://creativecommons.org/licenses/by/3.0/">
          <img
            alt="Creative Commons CC-BY-3.0 License"
            src="https://licensebuttons.net/l/by/3.0/80x15.png" />
        </a>
      </li>
    </ul>
  </footer>
);

export default Footer;
