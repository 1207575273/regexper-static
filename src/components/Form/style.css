@import url('../../globals.css');

:root {
  --control-gradient: var(--color-green) var(--gradient-green);
  --select-height: 2.8rem;
  --select-width: 12rem;
}

.form {
  margin: var(--spacing-margin) 0;
  overflow: hidden; /* Keep floated content in the box */

  & textarea {
    display: block;
    font-size: inherit;
    line-height: 1.5em;
    border: 0 none;
    outline: none;
    background: var(--color-tan);
    padding: 0 1rem;
    margin-bottom: var(--spacing-margin);
    width: 100% !important; /* "!important" to prevent user changing width */
    box-sizing: border-box;
    font-family: Consolas,Monaco,Lucida Console,Liberation Mono,DejaVu Sans Mono,Bitstream Vera Sans Mono,Courier New, monospace;
  }

  & textarea::placeholder {
    color: var(--color-brown);
  }

  & button {
    font-size: inherit;
    font-weight: bold;
    line-height: 2.8rem;
    width: 10rem;
    border: 0 none;
    background: var(--control-gradient);
    cursor: pointer;
    padding: 0;
    margin-right: 1rem;
  }
}

.actions {
  @apply --list-reset;
  float: right;

  @media (min-width: 700px) {
    @apply --inline-list;
    @apply --with-separator;
  }

  & svg {
    height: 1em;
    vertical-align: middle;
  }
}

.select {
  position: relative;
  vertical-align: bottom;
  display: inline-block;
  height: var(--select-height);
  width: var(--select-width);
  background: var(--control-gradient);
  overflow: hidden;

  & select {
    background: transparent;
    border: 0 none;
    font-size: inherit;
    height: var(--select-height);
    width: calc(var(--select-width) + 2rem);
  }

  & svg {
    position: absolute;
    top: calc((var(--select-height) - 24px) / 2);
    right: 0;
    pointer-events: none;
  }
}
