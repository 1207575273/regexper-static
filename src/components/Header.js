import React from 'react';
import { translate, Trans } from 'react-i18next';

import GithubIcon from 'feather-icons/dist/icons/github.svg';

const Header = () => (
  <header id="main" data-banner={ process.env.BANNER }>
    <h1>
      <a href="/">Regexper</a>
    </h1>

    <ul className="inline">
      <li><a href="https://github.com/javallone/regexper-static">
        <GithubIcon/><Trans>Source on GitHub</Trans>
      </a></li>
    </ul>
  </header>
);

export default translate()(Header);
export { Header };
