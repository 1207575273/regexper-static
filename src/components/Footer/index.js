import React from 'react';
import { translate, Trans } from 'react-i18next';

import style from './style.css';

const Footer = () => (
  <footer>
    <ul>
      <li>
        <Trans>Created by <a href="mailto:jeff.avallone@gmail.com">Jeff Avallone</a></Trans>
      </li>
      <li>
        <Trans i18nKey="Generated images licensed">
          Generated images licensed: <a rel="license external noopener noreferrer" target="_blank" href="http://creativecommons.org/licenses/by/3.0/">
            <img
              alt="Creative Commons CC-BY-3.0 License"
              src="https://licensebuttons.net/l/by/3.0/80x15.png" />
          </a>
        </Trans>
      </li>
      <li>
        <a href="/privacy.html"><Trans>Privacy Policy</Trans></a>
      </li>
    </ul>
    <div className={ style.buildId }>{ process.env.BUILD_ID }</div>
  </footer>
);

export default translate()(Footer);
export { Footer };
