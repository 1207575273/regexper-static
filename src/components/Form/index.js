import React from 'react';
import PropTypes from 'prop-types';
import { translate, Trans } from 'react-i18next';

import DownloadIcon from 'feather-icons/dist/icons/download.svg';
import LinkIcon from 'feather-icons/dist/icons/link.svg';
import ExpandIcon from 'feather-icons/dist/icons/chevrons-down.svg';

import style from './style.css';

class Form extends React.Component {
  handleSubmit = event => {
    event.preventDefault();
    this.props.onSubmit.call(this, {
      expr: this.textarea.value,
      syntax: this.syntax.value
    });
  }

  handleKeyPress = event => {
    if (event.charCode === 13 && event.shiftKey) {
      this.handleSubmit(event);
    }
  }

  textareaRef = textarea => this.textarea = textarea

  syntaxRef = syntax => this.syntax = syntax

  permalinkAction() {
    const { permalinkUrl } = this.props;

    if (!permalinkUrl) {
      return;
    }

    return <li>
      <a href={ permalinkUrl }><LinkIcon/><Trans>Permalink</Trans></a>
    </li>;
  }

  downloadActions() {
    const { downloadUrls } = this.props;

    if (!downloadUrls) {
      return;
    }

    return downloadUrls.map(({ url, filename, type, label }, i) => <li key={ i }>
      <a href={ url } download={ filename } type={ type }>
        <DownloadIcon/>{ label }
      </a>
    </li>);
  }

  render() {
    const { syntaxes, t } = this.props;

    return <div className={ style.form }>
      <form onSubmit={ this.handleSubmit }>
        <textarea
          ref={ this.textareaRef }
          onKeyPress={ this.handleKeyPress }
          autoFocus
          placeholder={ t('Enter regular expression to display') }></textarea>
        <button type="submit"><Trans>Display</Trans></button>
        <div className={ style.select }>
          <select ref={ this.syntaxRef }>
            { Object.keys(syntaxes).map(id => (
              <option value={ id } key={ id }>{ syntaxes[id] }</option>
            )) }
          </select>
          <ExpandIcon/>
        </div>
        <ul className={ ['inline', 'with-separator', style.actions].join(' ') }>
          { this.downloadActions() }
          { this.permalinkAction() }
        </ul>
      </form>

    </div>;
  }
}

Form.propTypes = {
  syntaxes: PropTypes.object,
  onSubmit: PropTypes.func,
  permalinkUrl: PropTypes.string,
  downloadUrls: PropTypes.array,
  t: PropTypes.func
};

export default translate()(Form);
export { Form };
