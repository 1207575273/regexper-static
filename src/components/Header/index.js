import React from 'react';
import { translate, Trans } from 'react-i18next';

import style from './style.css';
import GithubIcon from 'feather-icons/dist/icons/github.svg';

import LocaleSwitcher from 'components/LocaleSwitcher';

const Header = () => (
  <header className={ style.header } data-banner={ process.env.BANNER }>
    <h1>
      <a href="/">Regexper</a>
    </h1>

    <ul>
      <li><a href="https://github.com/javallone/regexper-static" rel="external noopener noreferrer" target="_blank">
        <GithubIcon/><Trans>Source on GitHub</Trans>
      </a></li>
      <li>
        <a href="/privacy.html"><Trans>Privacy Policy</Trans></a>
      </li>
      <li><LocaleSwitcher /></li>
    </ul>
  </header>
);

export default translate()(Header);
export { Header };
