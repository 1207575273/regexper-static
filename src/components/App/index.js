import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import URLSearchParams from 'url-search-params';

import style from './style.css';

import Form from 'components/Form';
import Message from 'components/Message';
import SVG from 'components/SVG';
import { demoImage } from 'devel';

const syntaxes = {
  js: 'JavaScript',
  pcre: 'PCRE'
};

class App extends React.PureComponent {
  state = {}

  image = React.createRef()

  componentDidMount() {
    window.addEventListener('hashchange', this.handleHashChange);
    this.handleHashChange();
  }

  componentWillUnmount() {
    window.removeEventListener('hashchange', this.handleHashChange);
  }

  async setSvgUrl() {
    try {
      const type = 'image/svg+xml';
      const blob = await this.image.current.svgUrl(type);

      this.setState({
        svgUrl: {
          url: URL.createObjectURL(blob),
          label: this.props.t('Download SVG'),
          filename: 'image.svg',
          type
        }
      });
    }
    catch (e) {
      console.error(e); // eslint-disable-line no-console
    }
  }

  async setPngUrl() {
    try {
      const type = 'image/png';
      const blob = await this.image.current.pngUrl(type);

      this.setState({
        pngUrl: {
          url: URL.createObjectURL(blob),
          label: this.props.t('Download PNG'),
          filename: 'image.png',
          type
        }
      });
    }
    catch (e) {
      console.error(e); // eslint-disable-line no-console
    }
  }

  handleSubmit = ({expr, syntax}) => {
    if (expr) {
      const params = new URLSearchParams({ syntax, expr });
      document.location.hash = params.toString();
    }
  }

  handleHashChange = () => {
    const query = document.location.hash.slice(1);
    const params = new URLSearchParams(query);
    const { expr, syntax } = (() => {
      if (params.get('syntax')) {
        return {
          syntax: params.get('syntax'),
          expr: params.get('expr')
        };
      } else {
        // Assuming old-style URL
        return {
          syntax: 'js',
          expr: query
        };
      }
    })();

    if (!expr) {
      return;
    }

    console.log(syntax, expr); // eslint-disable-line no-console
    this.setState({
      image: demoImage,
      permalinkUrl: document.location.toString(),
      syntax,
      expr
    }, async () => {
      await this.image.current.doReflow();
      this.setSvgUrl();
      this.setPngUrl();
    });
  }

  render() {
    const { svgUrl, pngUrl, permalinkUrl, syntax, expr, image } = this.state;
    const downloadUrls = [
      svgUrl,
      pngUrl
    ].filter(Boolean);

    return <React.Fragment>
      <Form
        syntaxes={ syntaxes }
        downloadUrls={ downloadUrls }
        permalinkUrl={ permalinkUrl }
        syntax={ syntax }
        expr={ expr }
        onSubmit={ this.handleSubmit }/>
      <Message type="error" heading="Sample Error">
        <p>Sample error message</p>
      </Message>
      <Message type="warning" heading="Sample Warning">
        <p>Sample warning message</p>
      </Message>
      { image && <div className={ style.render }>
        <SVG data={ image } ref={ this.image }/>
      </div> }
    </React.Fragment>;
  }
}

App.propTypes = {
  t: PropTypes.func
};

export default translate()(App);
export { App };
