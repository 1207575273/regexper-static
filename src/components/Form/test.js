import React from 'react';
import { shallow } from 'enzyme';

import { Form } from 'components/Form';
import { translate, I18nWrapper } from '__mocks__/i18n';

const syntaxes = {
  js: 'Javascript',
  pcre: 'PCRE'
};

describe('Form', () => {
  test('rendering', () => {
    const component = shallow(
      <Form t={ translate } syntaxes={ syntaxes }/>
    );
    expect(component).toMatchSnapshot();
  });

  test('rendering with download URLs');

  test('rendering with permalink URL');

  test('submitting form');

  test('submitting form with Shift+Enter');
});
