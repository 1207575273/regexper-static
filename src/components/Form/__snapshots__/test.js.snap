// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form rendering 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[Function]}
  >
    <textarea
      autoFocus={true}
      name="expr"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="translate(Enter regular expression to display)"
    />
    <button
      type="submit"
    >
      <Trans>
        Display
      </Trans>
    </button>
    <div
      className="select"
    >
      <select
        name="syntax"
        onChange={[Function]}
        value="js"
      >
        <option
          key="js"
          value="js"
        >
          Javascript
        </option>
        <option
          key="pcre"
          value="pcre"
        >
          PCRE
        </option>
      </select>
      <SvgMock />
    </div>
    <ul
      className="actions"
    />
  </form>
</div>
`;

exports[`Form rendering with download URLs 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[Function]}
  >
    <textarea
      autoFocus={true}
      name="expr"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="translate(Enter regular expression to display)"
    />
    <button
      type="submit"
    >
      <Trans>
        Display
      </Trans>
    </button>
    <div
      className="select"
    >
      <select
        name="syntax"
        onChange={[Function]}
        value="js"
      >
        <option
          key="js"
          value="js"
        >
          Javascript
        </option>
        <option
          key="pcre"
          value="pcre"
        >
          PCRE
        </option>
      </select>
      <SvgMock />
    </div>
    <ul
      className="actions"
    >
      <li
        key="0"
      >
        <a
          download="image.svg"
          href="#svg"
          type="image/svg+xml"
        >
          <SvgMock />
          <Trans>
            Download SVG
          </Trans>
        </a>
      </li>
      <li
        key="1"
      >
        <a
          download="image.png"
          href="#png"
          type="image/png"
        >
          <SvgMock />
          <Trans>
            Download PNG
          </Trans>
        </a>
      </li>
    </ul>
  </form>
</div>
`;

exports[`Form rendering with permalink URL 1`] = `
<div
  className="form"
>
  <form
    onSubmit={[Function]}
  >
    <textarea
      autoFocus={true}
      name="expr"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="translate(Enter regular expression to display)"
    />
    <button
      type="submit"
    >
      <Trans>
        Display
      </Trans>
    </button>
    <div
      className="select"
    >
      <select
        name="syntax"
        onChange={[Function]}
        value="js"
      >
        <option
          key="js"
          value="js"
        >
          Javascript
        </option>
        <option
          key="pcre"
          value="pcre"
        >
          PCRE
        </option>
      </select>
      <SvgMock />
    </div>
    <ul
      className="actions"
    >
      <li>
        <a
          href="#permalink"
        >
          <SvgMock />
          <Trans>
            Permalink
          </Trans>
        </a>
      </li>
    </ul>
  </form>
</div>
`;
